benchto-driver/src/main/java/com/teradata/benchto/driver/execution/ExecutionDriver.java

```diff
@@ -120,12 +125,15 @@ private void executeBenchmarks(List<Benchmark> benchmarks)
 
             executeHealthCheck(benchmark);
             benchmarkExecutionResults.add(benchmarkExecutionDriver.execute(benchmark, benchmarkOrdinalNumber++, benchmarks.size()));
+            benchmarkStatusReporter.processCompletedFutures();
         }
 
         List<BenchmarkExecutionResult> failedBenchmarkResults = benchmarkExecutionResults.stream()
                 .filter(benchmarkExecutionResult -> !benchmarkExecutionResult.isSuccessful())
                 .collect(toList());
 
+        benchmarkStatusReporter.awaitAllFutures(10, TimeUnit.MINUTES);
```

Comment:
This will be measured in minutes anyway. Maybe it's too little... In `ExecutionSynchronizer` there already is 20 minutes.